File: app/frontend/app/models/focus.js
import Ember from 'ember';
import DS from 'ember-data';

var Focus = DS.Model.extend({
  didLoad: function() {
  },
  locale: DS.attr('string'),
  pending: DS.attr('boolean'),
  permissions: DS.attr('raw'),
  title: DS.attr('string'),
  author: DS.attr('string'),
  category: DS.attr('string'),
  words: DS.attr('string'),
  all_words: DS.attr('string'),
  image_url: DS.attr('string'),
  source_url: DS.attr('string'),
  helper_words: DS.attr('string'),
  approved_users: DS.attr('raw'),
  all_words_capitalized: function() {
    return (this.get('all_words') || '').replace(/\bi\b/g, 'I');
  }.property('all_words')
});
// Brown Bear
// Cordurouy
// Guess How Much I Love You
// Dear Zoo
// But not the hippopotamus
// Are you my monster
// Go dog go
// Blue hat, Green hat
// The Gruffalo
//  There Was an Old Lady Who Swallowed a Fly
// Wheels on The Bus
// Goodnight Moon
// The Very Hungry Caterpillar
// Are You My Mother
// Add categories/topics to focus sets
// http://www.aacintervention.com/home/180009852/180009852/Images/repeated%20line%20books.pdf


export default Focus;

File: lib/json_api/focus.rb
module JsonApi::Focus
  extend JsonApi::Json
  
  TYPE_KEY = 'focus'
  DEFAULT_PAGE = 25
  MAX_PAGE =50
    
  def self.build_json(focus, args={})
    json = {}
    json['id'] = focus.ref_id
    json['locale'] = focus.locale || 'en'
    json['pending'] = !focus.id
    json['approved'] = !!focus.approved
    json['title'] = focus.title
    json['category'] = focus.category
    if focus.data
      json['all_words'] = focus.data['all_words']
      json['words'] = focus.data['words']
      json['author'] = focus.data['author']
      json['helper_words'] = focus.data['helper_words'] || focus.related_page_words
      json['source_url'] = focus.data['source_url']
      json['image_url'] = focus.data['image_url']
    end
    # https://shared.tarheelreader.org/shared/read/i-like-strawberries
    
    if args.has_key?(:permissions) && focus.data
      json['permissions'] = focus.permissions_for(args[:permissions])
      json['approved_users'] = focus.approved_users
    end
    json
  end
end
